### DISCUSSION TITLE
Handling of missing checks and agent_ids

### DISCUSSION THREAD

**Discussion by scmschmidt** [trento-project/wanda#127] [Q&A] [answered]
If none of the checks I request exists in the catalog, I get a 422 error with "no_checks_selected" as detail, but if at least one check exists, the missing checks are just left out in the response. Are you planning to add them to the response as missing or will the current behavior remain?

If an agent_id does not exists, Wanda will timeout (5 minutes) and the execution stays as "running" for that time. Shouldn't the existing agent_ids be known by Wanda so it can return with sort of a "Not found"?

**Comment by fabriziosestito** [comment, marked as answer]
@scmschmidt 

the "best effort" check execution is intentional. if a check does not exist, it doesn't get executed. 
_The dashboard user has no way to select non-existing checks so this is not planned to be enforced._

Also this simplifies backward compatibility with dashboard versions (e.g. a check is deprecated and removed from the catalog but still selected in the dashboard as stale data, we don't fail the whole execution).


_Similarly, the final user cannot select non-existing agents when starting an execution from the dashboard,_ so this is not enforced. We choose to not add the complexity of knowing which agents are up and running to wanda and leverage an async messaging system instead.

**Reply by scmschmidt** [reply]
In regards using the dashboard I agree. This acts as a safety layer. 

Not sure if we had discussed that i the past, but will the API be exposed to customers later? If so, at least missing agent_ids should not lead to a timeout.

**Reply by fabriziosestito** [reply]
@scmschmidt there is no plan to expose the execution API to the customers directly from wanda yet. If this feature will be requested we will probably expose an API from the discovery/dashboard layer.
